apiserver/facades/agent/uniter/uniter.go

```diff
@@ -150,7 +156,7 @@ func (u *UniterAPI) OpenedMachinePortRangesByEndpoint(ctx context.Context, args
 
 		result.Results[i].UnitPortRanges = make(map[string][]params.OpenUnitPortRangesByEndpoint)
 		for unitName, groupedPortRanges := range machPortRanges {
-			unitTag := names.NewUnitTag(unitName).String()
+			unitTag := names.NewUnitTag(unitName.String()).String()
```

Comment:
**suggestion:** I think we should make NewUnitTag take a UnitName, to avoid this double String() uglyness.

Reply:
It is a bit ugly. However, NewUnitTag if from an external `juju` package, so isn't worth changing. We're trying to minimise our usage of tags anyway, but there's still a long way to go. Once we have Mongo gone, we can make much greater progress towards this, but for now, imo, we should keep it as it is.

@SimonRichardson what do you think?

Reply:
I agree with @jack-w-shaw. The names package for all its warts should be relatively unchanged.

Reply:
Fair enough